packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -213,6 +213,7 @@ class ViewSettingsDialog extends UI5Element {
 			sortBy: [],
 			filters: [],
 		};
+		this._associatedSortByItem = null;
```

Comment:
This initialization doesn't work if we have predevined selected sort item, something like this:

`
ui5-view-settings-dialog id="vsd">

		<ui5-sort-item slot="sortItems" text="Name"></ui5-sort-item>
		<ui5-sort-item slot="sortItems" text="Position" **selected**></ui5-sort-item>
		<ui5-sort-item slot="sortItems" text="Company"></ui5-sort-item>
		<ui5-sort-item slot="sortItems" text="Department"></ui5-sort-item>
...
`
In case like that, initially _associatedSortByItem remains 'null', and immediate click on 'Confirm' (or if something is changed and then 'Reset' button is clicked) will return 'null' for sortByItem while there is something selected.

Please check this scenario.

Reply:
Yes, but is maybe this is expected behavior. This result come from confirm event - if not interacted or not select anything new maybe is not need to return element

Reply:
Whenever we fire an event, we need to ensure the event details are up to date and represent the current component state.
Usually, we just don't fire events, related to selection interactions, when there is no change in the selection.
However, in this case the event is called "confirm" (it's not called "change" or "selection-change"), so I guess it should be fired when the "Confirm" button is pressed, no matter if there is a change or not.
If this is true - we need to provide correct event details.